libraries/base/java-utils/Utils.java

```diff
@@ -123,6 +123,23 @@ public static String byteBufferToStr(ByteBuffer buffer)
         return new String(bytes, "UTF-8");
     }
 
+    public static ByteBuffer c_memcpy(ByteBuffer dest, ByteBuffer src, int size) {
+        src = src.duplicate();
+        ByteBuffer dest2 = dest.duplicate();
+        src.limit(src.position() + size);
```

Comment:
In `eta.runtime.io.MemoryManager`, there's a function called `bufSetLimit` that does the equivalent of what you're doing here, but it's more robust to changes in the MemoryManager.

Change `src.limit(src.position() + size)` to `bufSetLimit(src, size)`.